synthetic_data/dataset_generator.py

```diff
@@ -0,0 +1,93 @@
+from synthetic_data.distinct_generators.datetime_generator import random_datetimes
+from synthetic_data.distinct_generators.float_generator import random_floats
+from synthetic_data.distinct_generators.int_generator import random_integers
+from synthetic_data.distinct_generators.text_generator import random_string, random_text
+
+
+def convert_data_to_df(
+    np_data: np.array,
+    path: Optional[str] = None,
+    index: bool = False,
+    column_names: Optional[List[str]] = None,
+) -> pd.DataFrame:
+    """
+    Convert np array to a pandas dataframe.
+
+    :param np_data: np array to be converted
+    :type np_data: numpy array
+    :param path: path to output a csv of the dataframe generated
+    :type path: str, None, optional
+    :param index: whether to include index in output to csv
+    :type path: bool, optional
+    :param column_names: The names of the columns of a dataset
+    :type path: List, None, optional
+    :return: a pandas dataframe
+    """
+    # convert array into dataframe
+    if not column_names:
+        column_names = [x for x in range(len(np_data))]
+    dataframe = pd.DataFrame.from_dict(dict(zip(column_names, np_data)))
+    # save the dataframe as a csv file
+    if path:
+        dataframe.to_csv(path, index=index, encoding="utf-8")
+    return dataframe
+
+
+def generate_dataset_by_class(
+    rng: Generator,
+    columns_to_generate: List[dict],
+    dataset_length: int = 100000,
+    path: Optional[str] = None,
+) -> pd.DataFrame:
+    """
+    Randomizes a dataset with a mixture of different data classes.
+
+    :param rng: the np rng object used to generate random values
+    :type rng: numpy Generator
+    :param columns_to_generate: Classes of data to be included in the dataset
+    :type columns_to_generate: List[dict], None, optional
+    :param dataset_length: length of the dataset generated, default 100,000
+    :type dataset_length: int, optional
+    :param path: path to output a csv of the dataframe generated
+    :type path: str, None, optional
+
+    :return: pandas DataFrame
+    """
+    gen_funcs = {
+        "integer": random_integers,
+        "float": random_floats,
+        "categorical": random_categorical,
+        "text": random_text,
+        "datetime": random_datetimes,
+        "string": random_string,
+    }
+
+    if not columns_to_generate:
+        logging.warning(
+            "columns_to_generate is empty, empty dataframe will be returned."
+        )
+        return pd.DataFrame()
+
+    dataset = []
+    column_names = []
+    for col in columns_to_generate:
+        col_ = copy.deepcopy(col)
+        col_generator = col_.pop("generator")
+        if col_generator not in gen_funcs:
+            raise ValueError(f"generator: {col_generator} is not a valid generator.")
+
+        col_generator_function = gen_funcs.get(col_generator)
+        dataset.append(col_generator_function(**col_, rng=rng, num_rows=dataset_length))
+        column_names.append(col_generator)
```

Comment:
In a future PR, we will want to allow there to be a `name` field so we can specify the actual column name rather than using the gen. or someway to pass in the names. This will be important for profiles.

maybe that means restructuring `columns_to_generate`: `{"generator": ..., "name": ..., "props": ...}`